rest/admin_api.go

```diff
@@ -1423,78 +1424,123 @@ func (h *handler) handleDeleteDB() error {
 	return nil
 }
 
-// raw document access for admin api
-
+// handleGetRawDoc returns the raw version of the stored document for diagnostic purposes.
+// It can optionally redact UserData in the sync metadata and allows a custom salt for correlating with collected logs.
 func (h *handler) handleGetRawDoc() error {
 	h.assertAdminOnly()
-	docid := h.PathVar("docid")
 
-	includeDoc, includeDocSet := h.getOptBoolQuery("include_doc", true)
-	redact, _ := h.getOptBoolQuery("redact", false)
-	salt := h.getQuery("salt")
+	var (
+		docID                     = h.PathVar("docid")
+		includeDoc, includeDocSet = h.getOptBoolQuery("include_doc", true)
+		redact                    = h.getBoolQuery("redact")
+		redactSalt                = h.getQuery("salt")
+	)
 
-	if redact && includeDoc && includeDocSet {
-		return base.HTTPErrorf(http.StatusBadRequest, "redact and include_doc cannot be true at the same time. "+
-			"If you want to redact you must specify include_doc=false")
+	if redact {
+		// cannot explicitly include doc and redact - it doesn't make sense since the doc body itself is considered UserData and eligible for redaction.
+		if includeDocSet && includeDoc {
+			return base.HTTPErrorf(http.StatusBadRequest, "redact and include_doc cannot be true at the same time. "+
+				"If you want to redact you must specify include_doc=false")
+		}
+		// if include_doc was not explicitly set, and we want to redact, then automatically set include_doc to false
+		includeDoc = false
+	} else {
+		if redactSalt != "" {
+			return base.HTTPErrorf(http.StatusBadRequest, "redactSalt can only be used when redact=true")
+		}
 	}
 
-	if redact && !includeDocSet {
-		includeDoc = false
+	// collect the set of xattrs to fetch that Sync Gateway is interested in.
+	xattrKeys := []string{
```

Comment:
I would also like to put revseqno here.

`Database.syncGlobalSyncMouRevSeqNoAndUserXattrKeys()` might be correct, but actually that doesn't include revseqno! 

RevSeqNo will be really helpful for debugging XDCR

I think using a common function means if we added a future xattr that we might be more likely to pick it up?

Reply:
👍 included full `$document` virtual xattr - agree with importance for 4.0 debugging